source/server/active_tcp_listener.h

```diff
@@ -71,48 +75,45 @@ class ActiveTcpListener final : public Network::TcpListenerCallbacks,
                       bool hand_off_restored_destination_connections, bool rebalanced) override;
 
   /**
-   * Remove and destroy an active connection.
-   * @param connection supplies the connection to remove.
+   * Create active connection from the server connection.
    */
-  void removeConnection(ActiveTcpConnection& connection);
+  void newActiveConnection(const Network::FilterChain& filter_chain,
+                           Network::ServerConnectionPtr server_conn_ptr,
+                           std::unique_ptr<StreamInfo::StreamInfo> stream_info) override;
 
   /**
-   * Create a new connection from a socket accepted by the listener.
+   * Return the active connections container attached with the given filter chain.
    */
-  void newConnection(Network::ConnectionSocketPtr&& socket,
-                     std::unique_ptr<StreamInfo::StreamInfo> stream_info);
+  ActiveConnections& getOrCreateActiveConnections(const Network::FilterChain& filter_chain);
 
   /**
-   * Return the active connections container attached with the given filter chain.
+   * Update the listener config. The follow up connections will see the new config. The existing
+   * connections are not impacted.
    */
-  ActiveConnections& getOrCreateActiveConnections(const Network::FilterChain& filter_chain);
+  void updateListenerConfig(Network::ListenerConfig& config);
 
   /**
-   * Schedule to remove and destroy the active connections which are not tracked by listener
-   * config. Caution: The connection are not destroyed yet when function returns.
+   * Schedule to remove and destroy the active connection owned by the filter chain.
    */
-  void
-  deferredRemoveFilterChains(const std::list<const Network::FilterChain*>& draining_filter_chains);
+  void deferRemoveFilterChain(const Network::FilterChain* filter_chain) override;
```

Comment:
I would probably just call this `removeFilterChain()` as I think the deferral part doesn't really matter in this context.